src/main/java/edu/iu/terracotta/service/app/impl/OutcomeServiceImpl.java

```diff
@@ -176,33 +171,46 @@ public void updateOutcome(Long outcomeId, OutcomeDto outcomeDto) throws TitleVal
     }
 
     @Override
-    public void saveAndFlush(Outcome outcomeToChange) { allRepositories.outcomeRepository.saveAndFlush(outcomeToChange); }
+    public void saveAndFlush(Outcome outcomeToChange) {
+        allRepositories.outcomeRepository.saveAndFlush(outcomeToChange);
+    }
 
     @Override
-    public void deleteById(Long id) { allRepositories.outcomeRepository.deleteByOutcomeId(id); }
+    public void deleteById(Long id) {
+        allRepositories.outcomeRepository.deleteByOutcomeId(id);
+    }
 
     @Override
-    public boolean outcomeBelongsToExperimentAndExposure(Long experimentId, Long exposureId, Long outcomeId){
+    public boolean outcomeBelongsToExperimentAndExposure(Long experimentId, Long exposureId, Long outcomeId) {
         return allRepositories.outcomeRepository.existsByExposure_Experiment_ExperimentIdAndExposure_ExposureIdAndOutcomeId(experimentId, exposureId, outcomeId);
     }
 
     @Override
     public List<OutcomePotentialDto> potentialOutcomes(Long experimentId) throws DataServiceException, CanvasApiException {
         Optional<Experiment> experiment = experimentService.findById(experimentId);
+        List<Assignment> assignmentList = allRepositories.assignmentRepository.findByExposure_Experiment_ExperimentId(experimentId);
         List<OutcomePotentialDto> outcomePotentialDtos = new ArrayList<>();
         if (experiment.isPresent()) {
             String canvasCourseId = StringUtils.substringBetween(experiment.get().getLtiContextEntity().getContext_memberships_url(), "courses/", "/names");
-            List<AssignmentExtended> assignmentExtendedList = canvasAPIClient.listAssignments(canvasCourseId,experiment.get().getPlatformDeployment());
-            for (AssignmentExtended assignmentExtended:assignmentExtendedList){
+            List<AssignmentExtended> assignmentExtendedList = canvasAPIClient.listAssignments(canvasCourseId, experiment.get().getPlatformDeployment());
+            for (AssignmentExtended assignmentExtended : assignmentExtendedList) {
+                List<Assignment> matched = assignmentList.stream().filter(x -> {
```

Comment:
This should also filter out this experiment's consent assignment. You can get that lmsAssignmentId from `experiment.getConsentDocument().getLmsAssignmentId()`.

Reply:
@machristie this is fixed